src/formatDate.js

```diff
@@ -7,8 +7,62 @@
  *
  * @returns {string}
  */
-function formatDate(date, fromFormat, toFormat) {
-  // write code here
+function formatDate(dateStr, fromFormat, toFormat) {
+  const formatMap = {
+    YYYY: (d) => d,
+    YY: (d) => d.slice(-2),
+    MM: (d) => d.padStart(2, '0'),
+    DD: (d) => d.padStart(2, '0'),
```

Comment:
The 'MM' and 'DD' format functions use 'padStart' which is unnecessary since the input is assumed to be correctly formatted according to the task description. Consider removing 'padStart' for these formats.